feature_engine/discretisation/target_mean.py

```diff
@@ -0,0 +1,213 @@
+import warnings
+from typing import Dict, List, Optional, Union
+
+import pandas as pd
+
+from feature_engine.discretisation.base_discretiser import BaseDiscretiser
+from feature_engine._docstrings.methods import (
+    _fit_not_learn_docstring,
+    _fit_transform_docstring
+)
+from feature_engine._docstrings.fit_attributes import (
+    _variables_attribute_docstring,
+    _feature_names_in_docstring,
+    _n_features_in_docstring,
+)
+from feature_engine._docstrings.substitute import Substitution
+from feature_engine.dataframe_checks import (
+    _check_contains_inf,
+    _check_contains_na,
+    _check_X_matches_training_df,
+    check_X,
+)
+from feature_engine.discretisation import (
+    ArbitraryDiscretiser,
+    EqualFrequencyDiscretiser,
+    EqualWidthDiscretiser
+)
+from feature_engine.encoding import MeanEncoder
+from feature_engine.tags import _return_tags
+from feature_engine.variable_manipulation import (
+    _check_input_parameter_variables,
+    _find_or_check_numerical_variables,
+)
+
+from sklearn.pipeline import Pipeline
+
+@Substitution(
+    return_objects=BaseDiscretiser._return_object_docstring,
+    return_boundaries=BaseDiscretiser._return_boundaries_docstring,
+    binner_dict_=BaseDiscretiser._binner_dict_docstring,
+    transform=BaseDiscretiser._transform_docstring,
+    variables_=_variables_attribute_docstring,
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit=_fit_not_learn_docstring,
+    fit_transform=_fit_transform_docstring,
+)
+class TargetMeanDiscretiser(BaseDiscretiser):
+    """
+
+    Parameters
+    ----------
+    binning_dict: dict
+        The dictionary with the variable to interval limits pairs.
+
+    {return_object}
+
+    {return_boundaries}
+
+    errors: string, default='ignore'
+        Indicates what to do when a value is outside the limits indicated in the
+        'binning_dict'. If 'raise', the transformation will raise an error.
+        If 'ignore', values outside the limits are returned as NaN
+        and a warning will be raised instead.
+
+    Attributes
+    ----------
+    {variables_}
+
+    {binner_dict_}
+
+
+
+    {feature_names_in_}
+
+    {n_features_in_}
+
+    Methods
+    -------
+    {fit}
+
+    {fit_transform}
+
+    {transform}
+
+    See Also
+    --------
+    pandas.cut
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        bins: int = 5,
+        strategy: str = "equal_frequency",
+        binning_dict: Dict[Union[str, int], List[Union[str, int]]] = None,
```

Comment:
we do not need this parameter

Reply:
Are you referring to `binning_dict`? Yes, makes sense, given that we're removing the ArbitraryDiscretiser.